internal/controller/controllers/agent_controller.go

```diff
@@ -440,13 +467,25 @@ func (r *AgentReconciler) updateStatus(ctx context.Context, log logrus.FieldLogg
 			agent.Status.ValidationsInfo = newValidationsInfo
 		}
 
-		if h.Progress != nil && h.Progress.CurrentStage != "" {
-			if isNoneDay2Rebooting, err = r.isDay2NonePlatformHostRebooting(ctx, agent, h); err != nil {
-				log.WithError(err).Errorf("Failed to find if agent %s/%s belongs to none platform cluster and is rebooting", agent.Namespace, agent.Name)
-				return ctrl.Result{RequeueAfter: defaultRequeueAfterOnError}, nil
+		// Determine whether or not CSRs should be automatically signed.
+		if cluster != nil {
+			if cluster.UserManagedNetworking == nil {
+				log.Error("UserManagedNetworking was not set, cannot determine whether or not to auto approve CSRs")
+				shouldAutoApproveCSRs = false
+			} else {
+				isNonePlatform := *(cluster.UserManagedNetworking)
```

Comment:
Can you open a Jira ticket that about the fact that `UserManagedNetworking` should be somehow reflected in the AgentClusterInstall object?

Also add a comment above this line that says "We look at the database cluster object to check for UserManagedNetworking rather than look at the equivalent AgentClusterInstall field because that field the latter does not get properly set in Single Node clusters"

Reply:
> Can you open a Jira ticket that about the fact that UserManagedNetworking should be somehow correctly reflected in the AgentClusterInstall object


hmm dont we have it? Isnt it something that @ori-amizur worked on? @filanov

Reply:
We have a UserManagedNetworking field on ACI but it's set by users and doesn't sync when the backend implicitly sets it in the database for SNOs

Reply:
https://issues.redhat.com/browse/MGMT-10311

Reply:
Wrote in the ticket that we can use webhooks to set default values in the spec.